CONTRIBUTING.md

```diff
@@ -1,32 +1,90 @@
+####INDEX
+* [Pull Requests](#pull-requests)
+* [Setting Up Your Environment](#setting-up-your-environment)
+  * [Assumptions](#assumptions)
+  * [Your Repository](#setting-up-your-repository)
+  * [Solution Dependencies](#setting-up-the-solution-dependencies)
+
 Pull Requests
 =============
 
+* Please reference [Style Guide](STYLEGUIDE.md)
 * All PRs must be related to an open issue
-* If a PR changes the scripting API, it must include all related changes to the documentation. 
+* If a PR changes the scripting API, it must include all related changes to the documentation.
+* Try to keep types that do not rely directly on KSP types in the `kOS.Safe` project
+* Try to include unit tests in `kOS.Safe.Test` for additions to `kOS.Safe`
+* Try to include kerboscript test scripts for new language features in `kerboscript_tests`
+* Do not commit changes to the `.sln` or `.csproj` other than to add new source
+  files or new *required* references.
+* Do not commit files to the `Resources` folder (or its subfolders) that are not
+  *required* to be included in the redistributable zip file.
+* When updating your pull requests to the current develop branch, both `rebase`
+  and `merge` are acceptable.  If your pull request is knowingly being used as
+  the base for another pull request or another user, please avoid `rebase`.
 
 Setting Up Your Environment
 ===========================
+####Assumptions
+* `$KSP` is the full path to your Kerbal Space Program installation directory.
+  (i.e. `"C:\Program Files (x86)\Steam\SteamApps\common\Kerbal Space Program"`)
+* `$KOS` is the full path to your KOS git repository.
+* Other than examples of full paths for Windows, all paths are shown in this
+  guide using the unix directory seperator of `/`.  Please change all instances
+  of `/` and `\` to match the style of your development environment.
+* You already have a C# IDE and are familiar with how to use it to build a
+  Visual Studio solution.
```

Comment:
I think I'll call it a ".sln solution" because technically the solution file is not tied to a project being .net either (there is a way to use it for arduino projects, for example).